src/qseek/exporters/velest.py

```diff
@@ -2,34 +2,300 @@
+                        continue
+                    if (
+                        detection.phase == phase_s
+                        and observed.detection_value <= min_s_phase_confidence
+                    ):
+                        continue
+                    if (
+                        detection.traveltime_delay.total_seconds()
+                        > max_traveltime_delay
+                    ):
+                        continue
+                    observed_arrivals.append((receiver, detection))
+
+            countp, counts = self.export_phases_slim(
+                phase_file, event, observed_arrivals
+            )
+            self.n_picks_p += countp
+            self.n_picks_s += counts
+            n_earthquakes += 1
+        self.n_events = n_earthquakes
+
+        # export control file
+        control_file = outdir / "velest.cmn"
+        control_file_parameters = VelestControlFile(
+            ref_lat=search.octree.location.lat,
+            ref_lon=-search.octree.location.lon,
+            n_earthquakes=n_earthquakes,
+            max_distance_station=200,
+            min_depth=-0.2,
+            allow_low_velocity=False,
+            velocity_damping=1.0,
+            station_correction_damping=0.1,
+            use_elevation=False,
+            use_station_correction=False,
+            model_file="model.mod",
+            phase_file="phase_velest.pha",
+            mainout_file="main.out",
+            outcheck_file="log.out",
+            finalcnv_file="final.cnv",
+            stacorrection_file="stacor.dat",
+            isingle=False,
+            vp_vs_ratio=1.65,
+            iteration_number=99,
+            invertratio=0,
+        )
+        control_file_parameters.write_config_file(control_file)
+        # export velocity model file
+        dep = search.ray_tracers.root[0].earthmodel.layered_model.profile("z")
+        vp = search.ray_tracers.root[0].earthmodel.layered_model.profile("vp")
+        vs = search.ray_tracers.root[0].earthmodel.layered_model.profile("vs")
+        dep_velest = []
+        vp_velest = []
+        vs_velest = []
+        for i, d in enumerate(dep):
+            if float(d) / 1000 not in dep_velest:
+                dep_velest.append(float(d) / 1000)
+                vp_velest.append(float(vp[i]) / 1000)
+                vs_velest.append(float(vs[i]) / 1000)
+        velmod_file = outdir / "model.mod"
+        self.make_velmod_file(velmod_file, vp_velest, vs_velest, dep_velest)
 
         export_info = outdir / "export_info.json"
         export_info.write_text(self.model_dump_json(indent=2))
-
         return outdir
+
+    def export_phases_slim(
+        self,
+        outfile: Path,
+        event: EventDetection,
+        observed_arrivals: list[tuple[Receiver, PhaseDetection]],
+    ):
+        mag = event.magnitude.average if event.magnitude is not None else 0.0
+        lat = event.effective_lat
+        lon = event.effective_lon
+        if lat < 0:
```

Comment:
Create a helper function which creates the coordinate string, since this is used in different parts of the code.

```py
def velst_location(location: Location) -> str:
```